File: android/src/main/java/com/getcapacitor/community/classes/events/SentEvent.java
package com.getcapacitor.community.classes.events;

import android.util.Base64;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.getcapacitor.JSObject;
import java.util.UUID;

public class SentEvent extends MessageIDEvent {

    @Nullable
    String peerID;

    public SentEvent(@NonNull UUID messageID, @Nullable String peerID) {
        super(messageID);
        this.peerID = peerID;
    }

    @NonNull
    public JSObject toJSObject() {
        JSObject result = super.toJSObject();

        if (peerID != null) {
            result.put("peerID", peerID);
        }

        return result;
    }
}

File: ios/Sources/BitchatPlugin/Classes/Events/SentEvent.swift
import Foundation
import Capacitor

@objc public class SentEvent: MessageIDEvent {
    let peerID: String?

    init(_ messageID: UUID, _ peerID: String?) {
        self.peerID = peerID

        super.init(messageID)
    }

    override public func toJSObject() -> JSObject {
        var result = super.toJSObject()

        if let peerID = peerID {
            result["peerID"] = peerID
        }

        return result
    }
}
